src/build.rs

```diff
@@ -310,22 +318,22 @@ impl Build {
 
         if !self.disable_zig_linker {
             // setup zig as linker
-            if let Some(target) = self.target.as_ref() {
-                let rustc_meta = rustc_version::version_meta()?;
-                let host_target = &rustc_meta.host;
+            let rustc_meta = rustc_version::version_meta()?;
+            let host_target = &rustc_meta.host;
+            for (i, full_target) in rust_targets.iter().enumerate() {
                 // we only setup zig as linker when target isn't exactly the same as host target
-                if host_target != target {
-                    if let Some(rust_target) = rust_target {
-                        let env_target = rust_target.to_uppercase().replace('-', "_");
-                        let (zig_cc, zig_cxx) = prepare_zig_linker(target)?;
+                if host_target != full_target {
+                    if let Some(rust_target) = rust_targets.get(i) {
+                        let env_target = rust_target.replace('-', "_");
+                        let (zig_cc, zig_cxx) = prepare_zig_linker(full_target)?;
                         if is_mingw_shell() {
                             let zig_cc = zig_cc.to_slash_lossy();
-                            build.env("TARGET_CC", &zig_cc);
-                            build.env("TARGET_CXX", &zig_cxx.to_slash_lossy());
+                            build.env(format!("CC_{}", env_target), &zig_cc);
+                            build.env(format!("CXX_{}", env_target), &zig_cxx.to_slash_lossy());
                             build.env(format!("CARGO_TARGET_{}_LINKER", env_target), &zig_cc);
```

Comment:
this one needs upper case.

Reply:
This is a messy ride, I'm not that familiar with the cargo env variables, sorry for that. pushing right now.